sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/models/_models.py

```diff
@@ -957,30 +957,46 @@ def _to_generated_patch(
             if series_detection else None
         )
 
+class DataFeedSource(dict):
+    """DataFeedSource base class
+
+    :param data_source_type: Required. data source type.Constant filled by server.  Possible values
+     include: "AzureApplicationInsights", "AzureBlob", "AzureCosmosDB", "AzureDataExplorer",
+     "AzureDataLakeStorageGen2", "AzureEventHubs", "AzureLogAnalytics", "AzureTable", "InfluxDB",
+     "MongoDB", "MySql", "PostgreSql", "SqlServer".
+    :type data_source_type: str or ~azure.ai.metricsadvisor.models.DataSourceType
+    :keyword authentication_type: authentication type for corresponding data source. Possible values
+     include: "Basic", "ManagedIdentity", "AzureSQLConnectionString", "DataLakeGen2SharedKey",
+     "ServicePrincipal", "ServicePrincipalInKV". Default is "Basic".
+    :paramtype authentication_type: str or ~azure.ai.metricsadvisor.models.DataSourceAuthenticationType
+    :keyword str credential_id: The credential entity id.
+    """
+    def __init__(self, data_source_type, **kwargs):
+        # type: (str, **Any) -> None
+        super(DataFeedSource, self).__init__(data_source_type=data_source_type, **kwargs)
+        self.data_source_type = data_source_type
+        self.authentication_type = kwargs.get("authentication_type", None)
+        self.credential_id = kwargs.get("credential_id", None)
 
-class AzureApplicationInsightsDataFeedSource(object):
+class AzureApplicationInsightsDataFeedSource(DataFeedSource):
     """AzureApplicationInsightsDataFeedSource.
 
-    :keyword str azure_cloud: Azure cloud environment.
-    :keyword str application_id: Azure Application Insights ID.
-    :keyword str api_key: API Key.
     :param str query: Required. Query.
-    :param authentication_type: authentication type for corresponding data source. Possible values
-     include: "Basic", "ManagedIdentity", "AzureSQLConnectionString", "DataLakeGen2SharedKey",
-     "ServicePrincipal", "ServicePrincipalInKV". Default is "Basic".
-    :type authentication_type: str or ~azure.ai.metricsadvisor.models.DataSourceAuthenticationType
-    :keyword str credential_id: The credential entity id.
```

Comment:
IIRC, you might still have to have these docstrings in this class. I don't remember if sphinx will pick them up in the base class.